File: src-tauri/src/cli/ui.rs
use crate::cli::detector::ProjectTemplate;
use dialoguer::MultiSelect;
use std::path::Path;

/// Show preview of what will be created
pub fn show_preview(group_name: &str, directory: &Path, projects: &[ProjectTemplate]) {
    println!("\n╔════════════════════════════════════════════════════════╗");
    println!("║           PREVIEW: Group to be created                 ║");
    println!("╚════════════════════════════════════════════════════════╝");
    println!("  Group Name: {}", group_name);
    println!("  Directory:  {}", directory.display());
    println!("  Projects:   {}", projects.len());
    println!("  ───────────────────────────────────────────────────────");

    for (i, project) in projects.iter().enumerate() {
        println!("  {}. {}", i + 1, project.name);
        println!("     └─ {}", project.description);
        println!("     └─ $ {}", project.command);
        if i < projects.len() - 1 {
            println!();
        }
    }
    println!("  ═══════════════════════════════════════════════════════\n");
}

/// Prompt user for project selection using dialoguer
pub fn prompt_project_selection(
    projects: &[ProjectTemplate],
) -> Result<Vec<usize>, Box<dyn std::error::Error>> {
    if projects.is_empty() {
        return Ok(Vec::new());
    }

    let items: Vec<String> = projects
        .iter()
        .map(|p| format!("{} - {}", p.name, p.description))
        .collect();

    let defaults: Vec<bool> = vec![true; projects.len()];

    let selection = MultiSelect::new()
        .with_prompt("Select projects to include (use arrow keys to navigate, space to select/deselect, enter to confirm)")
        .items(&items)
        .defaults(&defaults)
        .interact()?;

    Ok(selection)
}

/// Prompt user for confirmation
pub fn prompt_confirmation(message: &str) -> Result<bool, Box<dyn std::error::Error>> {
    use dialoguer::Confirm;

    let confirmed = Confirm::new()
        .with_prompt(message)
        .default(true)
        .interact()?;

    Ok(confirmed)
}

File: src-electron/cli/ui.ts
/**
 * CLI UI utilities for interactive prompts
 */

import { checkbox, confirm } from '@inquirer/prompts';
import * as path from 'path';
import type { ProjectTemplate } from './types';

export function showPreview(groupName: string, directory: string, projects: ProjectTemplate[]): void {
  console.log('\n╔════════════════════════════════════════════════════════╗');
  console.log('║           PREVIEW: Group to be created                 ║');
  console.log('╚════════════════════════════════════════════════════════╝');
  console.log(`  Group Name: ${groupName}`);
  console.log(`  Directory:  ${directory}`);
  console.log(`  Projects:   ${projects.length}`);
  console.log('  ───────────────────────────────────────────────────────');

  for (let i = 0; i < projects.length; i++) {
    const project = projects[i];
    console.log(`  ${i + 1}. ${project.name}`);
    console.log(`     └─ ${project.description}`);
    console.log(`     └─ $ ${project.command}`);
    if (i < projects.length - 1) {
      console.log();
    }
  }
  console.log('  ═══════════════════════════════════════════════════════\n');
}

export async function promptProjectSelection(projects: ProjectTemplate[]): Promise<number[]> {
  if (projects.length === 0) {
    return [];
  }

  const choices = projects.map((p, index) => ({
    name: `${p.name} - ${p.description}`,
    value: index,
    checked: true
  }));

  try {
    const selected = await checkbox({
      message: 'Select projects to include (use arrow keys to navigate, space to select/deselect, enter to confirm):',
      choices
    });

    return selected.sort((a, b) => a - b);
  } catch {
    return [];
  }
}

export async function promptConfirmation(message: string): Promise<boolean> {
  try {
    return await confirm({
      message,
      default: true
    });
  } catch {
    return false;
  }
}
